### ISSUE TITLE
Coordinator View - Add a 'Reviewer' in a Course Evaluation

### ISSUE THREAD

**Issue by Mechsmith** [uwasystemhealth/IndEAAv2#13]
## What needs to be done? 
- Each Course Evaluation should be able to have reviewers added to the evaluation. This requires an option to select another user and grant them 'Reviewer' permissions.
- Adding a reviewer creates a 'Review' element in the database (empty except for the 'Reviewer' field)
- If that reviewer doesn't exist yet in the database, it creates a new user with reviewer permissions
- Part of coordinator view
- This issue is required in order to complete the rest of the reviewer pages
- Requires #12 before doing this ticket.
- Refers to this modal in IndEAAv1:
![Screen Shot 2022-06-24 at 1 19 09 pm](https://user-images.githubusercontent.com/64839317/175467518-8f5791a8-f22d-4b79-9225-071e0e7442b6.png)

## Other
Refers to 22 from our discussion
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/64839317/175467014-c49c6ba1-7b43-4711-b1d0-322c61b41227.png)

**Comment by MouseAndKeyboard** [comment]
In what way does this issue depend on #12 ?

**Comment by frinzekt** [comment]
> In what way does this issue depend on #12 ?
@MouseAndKeyboard , it does not... complete separate issue.

As far as this is concerned, from the implementation from #47. The frontend just needs to POST a value to `/api/v1/review/` with the `course_evaluation` (id) in the payload and an email (we can overwrite the create method two do a lookup of a user, or create the user for the `reviewer` field) . This will create a new entry of `Review`.

For any permissions such as `IsReviewOwnerAllOrCoordinatorReadOnly`... code seen below
```
class IsReviewOwnerAllOrCoordinatorReadOnly(permissions.BasePermission):
    """
    Usecase:
    - As a coordinator, I would like to create/delete a Course Review (to indicate that someone is appointed/removed as a reviewer)
    - As a coordinator, I would like to open the submission
    - As a reviewer (owner), I will be able to update my review
    - As a review (not owner), I am not able to do anything
    - As anonymous, I am not able to do anything
    """

    def has_object_permission(self, request, view, obj):
        # A coordinator can do anything
        if request.user in obj.course_evaluation.coordinators.all():
            return True

        # Reviewer in this context is the associated reviewer in the object
        return request.user == obj.reviewer
```

This one allows us to do assertions of review.

In the case of displaying all the reviewers... We can add that as a serializer as part of `CourseEvaluationDetailSerializer` probably something like `reviews = ReviewSerializer(many=True, read_only=True)`.

From a frontend POV (might be good to know @chayG5). We can revise the original IndEAA design, and avoid having the the reviewer in the `general` page, but have it in the `Review` panel. This will centralise anything that has to do with managing "reviewer" and "reviews" in one panel.

**Comment by frinzekt** [comment]
I am just thinking closely to this...

## The problem
To link a user, we need to create a user first. Something I was experimenting in `django-allauth` is what if we create the user first and then whatever social login provider, just picks that up as login.

It looks like I am missing something, because our current setup in #29 is having issue if a user already exist with the same email.

## What we can do

Looking at  the `django-allauth` code, we can do this:

```python
from allauth.account.models import EmailAddress
from django.contrib.auth.models import User

... somewhere we are going to create a user
user = User.objects.create(...)
email = EmailAddress.objects.create(user=user, ...)
```

This will enable us to simulate as if the user already has a social account provider ready

## Others
This means that as much as possible, we should not be creating accounts via django admin.
